src/common/options.cc

```diff
@@ -17,28 +17,91 @@
 // rbd feature validation
 #include "librbd/Features.h"
 
+ostream& operator<<(ostream& os, std::chrono::seconds secs) {
+  using namespace std::chrono;
+  using days_t = duration<int, std::ratio<3600 * 24>>;
+  bool printed = false;
+  if (auto d = duration_cast<days_t>(secs); d.count()) {
+    os << d.count() << " days";
+    secs -= d;
+    printed = true;
+  }
+  if (auto h = duration_cast<hours>(secs); printed || h.count()) {
+    if (printed) {
+      os << ", ";
+    }
+    os << h.count() << " hours";
+    secs -= h;
+    printed = true;
+  }
+  if (auto m = duration_cast<minutes>(secs); printed || m.count()) {
+    if (printed) {
+      os << ", ";
+    }
+    os << m.count() << " minutes";
+    secs -= m;
+    printed = true;
+  }
+  if (auto s = duration_cast<seconds>(secs); printed || s.count()) {
+    if (printed) {
+      os << ", ";
+    }
+    os << s.count() << " seconds";
+  }
+  return os;
+}
+
+namespace {
+class printer : public boost::static_visitor<> {
+  ostream& out;
+public:
+  explicit printer(ostream& os)
+    : out(os) {}
+  template<typename T>
+  void operator()(const T& v) const {
+    out << v;
+  }
+  void operator()(const Option::size_t& v) const {
+    cout << v.value;
+  }
+};
+}
+
+ostream& operator<<(ostream& os, const Option::value_t& v) {
+  printer p{os};
+  v.apply_visitor(p);
+  return os;
+}
 
 void Option::dump_value(const char *field_name,
     const Option::value_t &v, Formatter *f) const
 {
   if (boost::get<boost::blank>(&v)) {
     // This should be nil but Formatter doesn't allow it.
     f->dump_string(field_name, "");
-  } else if (type == TYPE_UINT) {
-    f->dump_unsigned(field_name, boost::get<uint64_t>(v));
-  } else if (type == TYPE_INT) {
-    f->dump_int(field_name, boost::get<int64_t>(v));
-  } else if (type == TYPE_STR) {
-    f->dump_string(field_name, boost::get<std::string>(v));
-  } else if (type == TYPE_FLOAT) {
-    f->dump_float(field_name, boost::get<double>(v));
-  } else if (type == TYPE_BOOL) {
-    f->dump_bool(field_name, boost::get<bool>(v));
-  } else {
-    f->dump_stream(field_name) << v;
+    return;
+  }
+  switch (type) {
+  case TYPE_INT:
+    f->dump_int(field_name, boost::get<int64_t>(v)); break;
+  case TYPE_UINT:
+    f->dump_unsigned(field_name, boost::get<uint64_t>(v)); break;
+  case TYPE_STR:
+    f->dump_string(field_name, boost::get<std::string>(v)); break;
+  case TYPE_FLOAT:
+    f->dump_float(field_name, boost::get<double>(v)); break;
+  case TYPE_BOOL:
+    f->dump_bool(field_name, boost::get<bool>(v)); break;
+  default:
+    f->dump_stream(field_name) << v; break;
```

Comment:
Is the call to dump_stream here going to end up calling the string-izing `operator<<` for TYPE_SECS?  We probably want to output those as integers, given that the Formatter output is meant for machine consumption

Reply:
okay, i will drop `ostream& operator<<(ostream& os, std::chrono::seconds secs) ` also.